services/outputhost/replicaconnection.go

```diff
@@ -382,20 +364,26 @@ func (conn *replicaConnection) writeCreditsPump() {
 		if numMsgsRead > creditBatchSize {
 			select {
 			case credits := <-conn.creditNotifyCh: // this is the common credit channel from redelivery cache
-				conn.utilSendCredits(credits, &numMsgsRead, &totalCreditsSent)
+				if err := sendCreditsToStore(credits); err != nil {
+					return
+				}
+				numMsgsRead -= credits
```

Comment:
Let's move this within the sendCreditsToStore() so that we don't need to decrement this every single time. You can rename this numMsgsRead as well..